### PR TITLE
resource/aws_autoscaling_policy: Add support for target_tracking_configuration

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
Support: #1156 
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSAutoscalingPolicy_TargetTrack -timeout 120m
=== RUN   TestAccAWSAutoscalingPolicy_TargetTrack_Predefined
--- PASS: TestAccAWSAutoscalingPolicy_TargetTrack_Predefined (87.74s)
=== RUN   TestAccAWSAutoscalingPolicy_TargetTrack_Custom
--- PASS: TestAccAWSAutoscalingPolicy_TargetTrack_Custom (82.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	169.792s
```

`adjustment_type` isn't supported if the policy type is TargetTrackingScaling so I changed Optional

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
What is the status?

File: aws/resource_aws_autoscaling_policy_test.go [hunk 1]
```diff
@@ -375,3 +415,79 @@ resource "aws_autoscaling_policy" "foobar_simple" {
 }
 `, name, name, name)
 }
+
+func testAccAwsAutoscalingPolicyConfig_TargetTracking_Predefined(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_launch_configuration" "test" {
+  name          = "tf-test-%s"
+  image_id      = "ami-21f78e11"
```
Comment (radeksimko) [inline aws/resource_aws_autoscaling_policy_test.go]: Nitpick: Do you mind using the AMI data source here instead, so we don't need to keep updating the ID over time as often?

File: aws/resource_aws_autoscaling_policy_test.go [hunk 2]
```diff
@@ -375,3 +415,79 @@ resource "aws_autoscaling_policy" "foobar_simple" {
...
+  instance_type = "t1.micro"
+}
+
+resource "aws_autoscaling_group" "test" {
+  availability_zones        = ["us-west-2a"]
```
Comment (radeksimko) [inline aws/resource_aws_autoscaling_policy_test.go]: Do you mind using the AZ data source here to make the test region-agnostic?
https://www.terraform.io/docs/providers/aws/d/availability_zones.html

File: aws/resource_aws_autoscaling_policy_test.go [hunk 3]
```diff
@@ -375,3 +415,79 @@ resource "aws_autoscaling_policy" "foobar_simple" {
 }
 `, name, name, name)
 }
+
+func testAccAwsAutoscalingPolicyConfig_TargetTracking_Predefined(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_launch_configuration" "test" {
+  name          = "tf-test-%s"
+  image_id      = "ami-21f78e11"
+  instance_type = "t1.micro"
```
Comment (radeksimko) [inline aws/resource_aws_autoscaling_policy_test.go]: AFAIK `t1` family of instances has been deprecated - do you mind changing this to any low-price size of current family? e.g. `t2.micro`? https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/concepts_micro_instances.html

File: aws/resource_aws_autoscaling_policy_test.go [hunk 4]
```diff
@@ -375,3 +415,79 @@ resource "aws_autoscaling_policy" "foobar_simple" {
...
+}
+`, rName, rName, rName)
+}
+
+func testAccAwsAutoscalingPolicyConfig_TargetTracking_Custom(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_launch_configuration" "test" {
+  name          = "tf-test-%s"
+  image_id      = "ami-21f78e11"
+  instance_type = "t1.micro"
```
Comment (radeksimko) [inline aws/resource_aws_autoscaling_policy_test.go]: Likewise - as above ^

File: aws/resource_aws_autoscaling_policy_test.go [hunk 5]
```diff
@@ -375,3 +415,79 @@ resource "aws_autoscaling_policy" "foobar_simple" {
...
+}
+
+resource "aws_autoscaling_group" "test" {
+  availability_zones        = ["us-west-2a"]
```
Comment (radeksimko) [inline aws/resource_aws_autoscaling_policy_test.go]: Likewise - as above ^

**Comment by atsushi-ishibashi** [comment]
@radeksimko Ok👍
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSAutoscalingPolicy_TargetTrack* -timeout 120m
=== RUN   TestAccAWSAutoscalingPolicy_TargetTrack_Predefined
--- PASS: TestAccAWSAutoscalingPolicy_TargetTrack_Predefined (110.02s)
=== RUN   TestAccAWSAutoscalingPolicy_TargetTrack_Custom
--- PASS: TestAccAWSAutoscalingPolicy_TargetTrack_Custom (105.76s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	215.812s
```

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**